docker/package/tezos_setup_wizard.py

```diff
@@ -422,25 +426,32 @@ def get_key_mode_query(modes):
     validator=Validator([required_field_validator, filepath_validator]),
 )
 
+derivation_path_query = Step(
+    id="derivation_path",
+    prompt="Provide derivation path for the key stored on the ledger.",
+    help="The format is '(bip25519|ed25519|secp256k1|P-256)/[0-9]+h/[0-9]+h'",
+    validator=Validator([required_field_validator, derivation_path_validator]),
+)
+
 # We define this step as a function since the corresponding step requires
 # tezos-node to be running and bootstrapped in order to gather the data
 # about the ledger-stored addresses, so it's called right before invoking
 # after the node was boostrapped
-def get_ledger_url_query(connected_ledgers, node_endpoint):
+def get_ledger_url_query(connected_ledgers, base_ledger_url, node_endpoint):
+    extra_options = ["Specify derivation path", "Go back"]
     return Step(
         id="ledger_url",
-        prompt="Provide the ledger URL for importing the baker secret key stored on a ledger.\n"
-        "You can choose one of the suggested options or provide your own ledger URL.",
-        help="Your ledger URL should look something like <ledger://<mnemonic>/ed25519/0'/1'>",
+        prompt="Select a key to import from " + base_ledger_url + ".\n"
+        "You can choose one of the suggested derivations or provide your own:",
+        help="'Specify derivation path' will ask a derivation path from you."
+        "'Go back' will return you back to the key type choice.",
         default=None,
-        options=ledger_urls_info(connected_ledgers, node_endpoint),
+        options=ledger_urls_info(connected_ledgers, base_ledger_url, node_endpoint)
+        + extra_options,
         validator=Validator(
             [
                 required_field_validator,
-                or_validator(
-                    enum_range_validator(connected_ledgers),
-                    ledger_url_validator,
-                ),
+                enum_range_validator(connected_ledgers + extra_options),
```

Comment:
I think there might be a problem here with validation: `connected_ledgers` is built up using `ledger_regex`, so it's still just full ledger URLs. `ledger_urls_info` will truncate the common base URL and only print derivation paths. However, the validator here still receives full URLs via `connected_ledgers`, so if the user types in an (already present) derivation instead of, say, `3`, it will refuse to validate.

Reply:
Well, the point is to allow only numeric options here, so that a user choose `Specify derivation path` and inputs derivation in the next prompt

Reply:
Yeah, I know, I'm talking about `enum_range_validator` accepting both numeric options and their written out counterparts. E.g. in the `How do you want to import the baker key` step it's possible to answer both `1` and `ledger`, but not `42` or `idk whatever`. Here, the prompt is for derivation paths, so from the expected behaviour it should be possible to choose `ed25519/0h/0h` by either typing it out or entering `2`. But the validator here accepts full ledger URLs, so `ed25519/0h/0h` would not be accepted, only `ledger://live-fossa-eager-walrus/ed25519/0h/0h`.

Reply:
Oh, okay, then something smarter is needed indeed

Reply:
As I mention in my original issue description, I don't think user should need to type algo name, just derivation path e.g. 0h/1h, (which btw can be empty and produces yet another set of addresses) or even accept just a pair of numbers types like 0/1 (not sure if those `h` and `'` sometimes used for derivation path need to be typed)

Reply:
I'm not sure if it's possible to provide an URL without the algo name to the `tezos-client` :thinking:

Reply:
to quote myself:

Reply:
Ah, okay, that makes sense